crates/blockchain/payload.rs

```diff
@@ -444,6 +468,120 @@ impl Blockchain {
+            *withdrawals_size += Self::L2_WITHDRAWAL_SIZE;
+        }
+        if Self::is_deposit_l2(&tx) {
+            *deposits_size += Self::L2_DEPOSIT_SIZE;
+        }
+        let modified_accounts_size = Self::calc_modified_accounts_size(context)?;
+        let current_state_diff_size =
+            8 /* version (u8) */ + *withdrawals_size + *deposits_size + modified_accounts_size;
+        dbg!(current_state_diff_size);
+        dbg!(&withdrawals_size);
+        dbg!(&deposits_size);
+        dbg!(modified_accounts_size);
+        if *withdrawals_size + *deposits_size + modified_accounts_size > 5000 {
+            // if *withdrawals_size + *deposits_size + modified_accounts_size > BYTES_PER_BLOB * 31 / 32 {
+            if Self::is_withdrawal_l2(&tx, receipt) {
+                *withdrawals_size -= Self::L2_WITHDRAWAL_SIZE;
+            }
+            if Self::is_deposit_l2(&tx) {
+                *deposits_size -= Self::L2_DEPOSIT_SIZE;
+            }
+            warn!("Exceeded blob size");
+            return Ok(false);
+        }
+        Ok(true)
+    }
+
+    fn is_withdrawal_l2(tx: &Transaction, receipt: &Receipt) -> bool {
+        // WithdrawalInitiated(address,address,uint256)
+        let withdrawal_event_selector: H256 =
+            H256::from_str("bb2689ff876f7ef453cf8865dde5ab10349d222e2e1383c5152fbdb083f02da2")
+                .unwrap();
+
+        match tx.to() {
+            TxKind::Call(to) if to == Self::COMMON_BRIDGE_L2_ADDRESS => {
+                return receipt.logs.iter().any(|log| {
+                    log.topics
+                        .iter()
+                        .any(|topic| *topic == withdrawal_event_selector)
+                });
+            }
+            _ => false,
+        }
+    }
+
+    fn is_deposit_l2(tx: &Transaction) -> bool {
+        matches!(tx, Transaction::PrivilegedL2Transaction(_tx))
+    }
+
+    fn calc_modified_accounts_size(context: &mut PayloadBuildContext) -> Result<usize, ChainError> {
+        // Starts from modified_accounts_len(u16)
+        let mut modified_accounts_size: usize = 16;
+        let account_updates = context
+            .vm
+            .get_state_transitions(context.payload.header.parent_hash)?;
+        for account_update in account_updates {
+            // r#type(u8) + address(H160)
+            modified_accounts_size += 8 + 160;
+            if account_update.info.is_some() {
+                modified_accounts_size += 256; // new_balance(U256)
+            }
+            if Self::has_new_nonce(&account_update, context)? {
+                modified_accounts_size += 16; // nonce_diff(u16)
+            }
+            if !account_update.added_storage.is_empty() {
+                modified_accounts_size += 16; // stoarge_len(u16)
+                for _storage in account_update.added_storage.iter() {
+                    modified_accounts_size += 256; // key(H256)
+                    modified_accounts_size += 256; // value(U256)
+                }
```

Comment:
```suggestion
                modified_accounts_size += account_update.added_storage.len() * 32 * 2
```
The 32 here are the 256bits